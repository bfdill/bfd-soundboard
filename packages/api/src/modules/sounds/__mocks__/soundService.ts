import { ISoundService, Sound } from '..'

export interface IMockSoundService extends ISoundService {
  setGetBySoundId: (result: any) => void
  setGetSounds: (result: any) => void
  setIsPathValid: (result: boolean) => void
  getBySoundIdMock: () => jest.Mock
  getSoundsMock: (result: any) => jest.Mock
  IsPathValidMock: (result: boolean) => jest.Mock
}

export const getMockSoundService = (): IMockSoundService => {
  const getBySoundId = jest.fn()
  const getSounds = jest.fn()
  const isPathValid = jest.fn()

  return {
    getBySoundId,
    getSounds,
    isPathValid,
    getBySoundIdMock: getBySoundId,
    getSoundsMock: getSounds,
    IsPathValidMock: isPathValid,
    setGetBySoundId: (result: Sound | undefined) =>
      getBySoundId.mockReturnValue(result),
    setGetSounds: (result: Sound[]) => getSounds.mockReturnValue(result),
    setIsPathValid: (result: boolean) => isPathValid.mockReturnValue(result)
  } as any
}
