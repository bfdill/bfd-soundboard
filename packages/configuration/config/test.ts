import { DeepPartial } from '@talkboy-9000/utils'

import { FullConfig } from '../src/configuration.types'

const config: DeepPartial<FullConfig> = {}

export default config
