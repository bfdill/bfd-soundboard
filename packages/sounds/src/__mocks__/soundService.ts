import { Sound } from '@talkboy-9000/models'
import { ISoundService } from '../sounds.types'

export interface IMockSoundService extends ISoundService {
  setGetBySoundId: (result: any) => void
  setGetSounds: (result: any) => void
  setIsPathValid: (result: boolean) => void
  getBySoundIdMock: () => jest.Mock
  getSoundsMock: (result: any) => jest.Mock
  IsPathValidMock: (result: boolean) => jest.Mock
}

export const getMockSoundService = (): IMockSoundService => {
  const getBySoundId = jest.fn()
  const getSounds = jest.fn()
  const isPathValid = jest.fn()

  return <IMockSoundService>{
    getBySoundId,
    getSounds,
    isPathValid,
    getBySoundIdMock: getBySoundId,
    getSoundsMock: getSounds,
    IsPathValidMock: isPathValid,
    setGetBySoundId: (result: Sound | undefined) =>
      getBySoundId.mockReturnValue(result),
    setGetSounds: (result: Sound[]) => getSounds.mockReturnValue(result),
    setIsPathValid: (result: boolean) => isPathValid.mockReturnValue(result)
  }
}
