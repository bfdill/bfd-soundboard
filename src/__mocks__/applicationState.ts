import { ApplicationState } from '../types'

export const mockApplicationState: ApplicationState = {
  correlationId: 'default_mock_correlation_id'
}
