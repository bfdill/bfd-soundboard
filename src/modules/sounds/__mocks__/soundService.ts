import { ISoundService, Sound } from '..'

export interface IMockSoundService extends ISoundService {
  setGetBySoundId: (result: Sound | undefined) => void
  setGetSounds: (result: Sound[]) => void
  setIsPathValid: (result: boolean) => void
  getBySoundIdMock: () => jest.Mock
  getSoundsMock: (result: any) => jest.Mock
  IsPathValidMock: (result: boolean) => jest.Mock
}

export const getMockSoundService = (): IMockSoundService => {
  const getBySoundId = jest.fn()
  const getSounds = jest.fn()
  const isPathValid = jest.fn()

  return {
    getBySoundId,
    getSounds,
    isPathValid,
    getBySoundIdMock: getBySoundId,
    getSoundsMock: getSounds,
    IsPathValidMock: isPathValid,
    setGetBySoundId: (result: Sound | undefined) =>
      getBySoundId.mockReturnValue(result),
    setGetSounds: (result: Sound[]) => getSounds.mockReturnValue(result),
    setIsPathValid: (result: boolean) => isPathValid.mockReturnValue(result)
  } as any
}
