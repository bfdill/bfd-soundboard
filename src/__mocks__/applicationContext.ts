import { mockApplicationState } from './applicationState'
import { IApplicationContext } from '../types'
import { getJestLogger } from '../modules/winston-jest/index.test'

export const getMockApplicationContext = (): IApplicationContext =>
  ({
    state: mockApplicationState
  } as any)
