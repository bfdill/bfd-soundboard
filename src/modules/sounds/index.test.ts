import { SoundService, PATH_TO_SOUNDS, ISoundService } from '.'
import * as sane from 'sane'
import { getJestLogger, IJestLogger } from '../winston-jest/index.test'
import { join } from 'path'
import { mockApplicationState } from '../../__mocks__/applicationState'

describe('modules -> sounds', () => {
  const saneOnMock = jest.fn()
  const saneWatcherMock = {
    on: saneOnMock
  }
  const saneFunctionMock = (_: string, _1?: sane.Options): sane.Watcher => {
    return saneWatcherMock as any
  }
  const jestLogger: IJestLogger = getJestLogger()
  let soundService: ISoundService
  let baseSoundService: SoundService

  beforeEach(() => {
    saneOnMock.mockReturnValue(saneWatcherMock)
    soundService = new SoundService(
      PATH_TO_SOUNDS,
      saneFunctionMock as any,
      jestLogger.logger
    )
    baseSoundService = soundService as SoundService
  })

  test('has known exports', () => {
    expect(Object.keys(require('.'))).toMatchSnapshot()
  })

  describe('addSound', () => {
    test('id = ordinal position by filename alpha', () => {
      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toHaveLength(0)
      baseSoundService.addSound('/tmp/fake_sound_0.mp3')

      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toHaveLength(1)
      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toMatchSnapshot()

      baseSoundService.addSound('/tmp/a_fake_sound_0.mp3')

      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toHaveLength(2)
      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toMatchSnapshot()

      jestLogger.callsMatchSnapshot()
    })
  })

  describe('createWatch', () => {
    test('wires add and delete on instantiation', () => {
      saneOnMock.mock.calls.forEach(call => {
        expect(call).toMatchSnapshot()
      })

      jestLogger.callsMatchSnapshot()
    })

    test('add, adds', () => {
      const addSoundSpy = jest.spyOn(baseSoundService, 'addSound')
      const fn = saneOnMock.mock.calls[0][1]
      fn('add test')
      expect(addSoundSpy).toHaveBeenCalledWith('add test')
      jestLogger.callsMatchSnapshot()
    })

    test('delete, removes', () => {
      const removeSoundSpy = jest.spyOn(baseSoundService, 'removeSound')
      const fn = saneOnMock.mock.calls[1][1]
      fn('remove test')
      expect(removeSoundSpy).toHaveBeenCalledWith('remove test')
      jestLogger.callsMatchSnapshot()
    })
  })

  describe('getBySoundId', () => {
    test.each([undefined, null, '', '0', 'dog'])(
      "getBySoundId('%s')",
      (soundId: string) => {
        baseSoundService.addSound('/tmp/fake_sound_0.mp3')
        jestLogger.transport.mock.mockReset()

        expect(
          soundService.getBySoundId(
            soundId,
            mockApplicationState,
            jestLogger.logger
          )
        ).toMatchSnapshot()
        jestLogger.callsMatchSnapshot()
      }
    )
  })

  describe('getSounds', () => {
    test('no sounds', () => {
      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toHaveLength(0)

      jestLogger.callsMatchSnapshot()
    })

    test('two sounds', () => {
      expect(
        soundService.getSounds(mockApplicationState, jestLogger.logger)
      ).toHaveLength(0)

      baseSoundService.addSound('/tmp/fake_sound_0.mp3')
      baseSoundService.addSound('/tmp/a_fake_sound.mp3')
      jestLogger.transport.mock.mockReset()

      const sounds = soundService.getSounds(
        mockApplicationState,
        jestLogger.logger
      )
      expect(sounds).toHaveLength(2)
      expect(sounds).toMatchSnapshot()

      jestLogger.callsMatchSnapshot()
    })
  })

  describe('isPathValid', () => {
    test.each([
      ['/tmp/invalid_af', false],
      [join(PATH_TO_SOUNDS, 'too_legit'), true]
    ])('validity test', (filename: string, expected) => {
      const actual = soundService.isPathValid(
        filename,
        mockApplicationState,
        jestLogger.logger
      )

      expect(actual).toEqual(expected)
    })
  })

  describe('removeSound', () => {
    beforeEach(() => {
      baseSoundService.addSound('a')
      baseSoundService.addSound('b')
      baseSoundService.addSound('c')
      jestLogger.transport.mock.mockReset()
    })

    test('unknown file, no action', () => {
      const expected = baseSoundService.getSounds(
        mockApplicationState,
        jestLogger.logger
      )
      jestLogger.transport.mock.mockReset()

      baseSoundService.removeSound('f')
      jestLogger.callsMatchSnapshot()

      const actual = baseSoundService.getSounds(
        mockApplicationState,
        jestLogger.logger
      )
      expect(actual).toEqual(expected)
      expect(actual).toHaveLength(3)
    })

    test.each(['a', 'b', 'c'])(
      'removes file and re-sorts remaining items',
      (filename: string) => {
        baseSoundService.removeSound(filename)
        jestLogger.callsMatchSnapshot()

        expect(
          baseSoundService.getSounds(mockApplicationState, jestLogger.logger)
        ).toMatchSnapshot()
      }
    )
  })
})
