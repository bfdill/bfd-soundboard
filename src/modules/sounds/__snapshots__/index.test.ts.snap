// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules -> sounds addSound id = ordinal position by filename alpha 1`] = `
Array [
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "0",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 2`] = `
Array [
  Object {
    "basename": "a_fake_sound_0.mp3",
    "filename": "/tmp/a_fake_sound_0.mp3",
    "id": "0",
  },
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "1",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 3`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 4`] = `
Array [
  Object {
    "level": "debug",
    "message": "addSound(/tmp/fake_sound_0.mp3)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"addSound(/tmp/fake_sound_0.mp3)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 5`] = `
Array [
  Object {
    "level": "silly",
    "message": "addSound(/tmp/fake_sound_0.mp3)",
    "sounds": Array [
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"addSound(/tmp/fake_sound_0.mp3)\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 6`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 7`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 8`] = `
Array [
  Object {
    "level": "debug",
    "message": "addSound(/tmp/a_fake_sound_0.mp3)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"addSound(/tmp/a_fake_sound_0.mp3)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 9`] = `
Array [
  Object {
    "level": "silly",
    "message": "addSound(/tmp/a_fake_sound_0.mp3)",
    "sounds": Array [
      Object {
        "basename": "a_fake_sound_0.mp3",
        "filename": "/tmp/a_fake_sound_0.mp3",
        "id": "0",
      },
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"addSound(/tmp/a_fake_sound_0.mp3)\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 10`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [
      Object {
        "basename": "a_fake_sound_0.mp3",
        "filename": "/tmp/a_fake_sound_0.mp3",
        "id": "0",
      },
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds addSound id = ordinal position by filename alpha 11`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [
      Object {
        "basename": "a_fake_sound_0.mp3",
        "filename": "/tmp/a_fake_sound_0.mp3",
        "id": "0",
      },
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound_0.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "addSound(add test)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"addSound(add test)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 2`] = `
Array [
  Object {
    "level": "silly",
    "message": "addSound(add test)",
    "sounds": Array [
      Object {
        "basename": "add test",
        "filename": "add test",
        "id": "0",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"addSound(add test)\\",\\"sounds\\":[{\\"filename\\":\\"add test\\",\\"id\\":\\"0\\",\\"basename\\":\\"add test\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds createWatch add, adds 3`] = `
Array [
  Object {
    "level": "debug",
    "message": "add",
    "path": "add test",
    Symbol(level): "debug",
    Symbol(message): "{\\"path\\":\\"add test\\",\\"message\\":\\"add\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch delete, removes 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(remove test)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(remove test)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch delete, removes 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "delete",
    "path": "remove test",
    Symbol(level): "debug",
    Symbol(message): "{\\"path\\":\\"remove test\\",\\"message\\":\\"delete\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds createWatch wires add and delete on instantiation 1`] = `
Array [
  "add",
  [Function],
]
`;

exports[`modules -> sounds createWatch wires add and delete on instantiation 2`] = `
Array [
  "delete",
  [Function],
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId()",
    "result": undefined,
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"getBySoundId()\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('0') 1`] = `
Object {
  "basename": "fake_sound_0.mp3",
  "filename": "/tmp/fake_sound_0.mp3",
  "id": "0",
}
`;

exports[`modules -> sounds getBySoundId getBySoundId('0') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(0)",
    "result": Object {
      "basename": "fake_sound_0.mp3",
      "filename": "/tmp/fake_sound_0.mp3",
      "id": "0",
    },
    Symbol(level): "debug",
    Symbol(message): "{\\"result\\":{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"fake_sound_0.mp3\\"},\\"message\\":\\"getBySoundId(0)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('dog') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('dog') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(dog)",
    "result": undefined,
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"getBySoundId(dog)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('null') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('null') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(null)",
    "result": undefined,
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"getBySoundId(null)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getBySoundId getBySoundId('undefined') 1`] = `undefined`;

exports[`modules -> sounds getBySoundId getBySoundId('undefined') 2`] = `
Array [
  Object {
    "level": "debug",
    "message": "getBySoundId(undefined)",
    "result": undefined,
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"getBySoundId(undefined)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds getSounds no sounds 1`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds getSounds two sounds 1`] = `
Array [
  Object {
    "basename": "a_fake_sound.mp3",
    "filename": "/tmp/a_fake_sound.mp3",
    "id": "0",
  },
  Object {
    "basename": "fake_sound_0.mp3",
    "filename": "/tmp/fake_sound_0.mp3",
    "id": "1",
  },
]
`;

exports[`modules -> sounds getSounds two sounds 2`] = `
Array [
  Object {
    "level": "silly",
    "message": "getSounds()",
    "sounds": Array [
      Object {
        "basename": "a_fake_sound.mp3",
        "filename": "/tmp/a_fake_sound.mp3",
        "id": "0",
      },
      Object {
        "basename": "fake_sound_0.mp3",
        "filename": "/tmp/fake_sound_0.mp3",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"getSounds()\\",\\"sounds\\":[{\\"filename\\":\\"/tmp/a_fake_sound.mp3\\",\\"id\\":\\"0\\",\\"basename\\":\\"a_fake_sound.mp3\\"},{\\"filename\\":\\"/tmp/fake_sound_0.mp3\\",\\"id\\":\\"1\\",\\"basename\\":\\"fake_sound_0.mp3\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds has known exports 1`] = `
Array [
  "soundServiceLogger",
  "PATH_TO_SOUNDS",
  "SoundService",
  "soundService",
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(a)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(a)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 2`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(a)",
    "sounds": Array [
      Object {
        "basename": "b",
        "filename": "b",
        "id": "0",
      },
      Object {
        "basename": "c",
        "filename": "c",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(a)\\",\\"sounds\\":[{\\"filename\\":\\"b\\",\\"id\\":\\"0\\",\\"basename\\":\\"b\\"},{\\"filename\\":\\"c\\",\\"id\\":\\"1\\",\\"basename\\":\\"c\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 3`] = `
Array [
  Object {
    "basename": "b",
    "filename": "b",
    "id": "0",
  },
  Object {
    "basename": "c",
    "filename": "c",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 4`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(b)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(b)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 5`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(b)",
    "sounds": Array [
      Object {
        "basename": "a",
        "filename": "a",
        "id": "0",
      },
      Object {
        "basename": "c",
        "filename": "c",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(b)\\",\\"sounds\\":[{\\"filename\\":\\"a\\",\\"id\\":\\"0\\",\\"basename\\":\\"a\\"},{\\"filename\\":\\"c\\",\\"id\\":\\"1\\",\\"basename\\":\\"c\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 6`] = `
Array [
  Object {
    "basename": "a",
    "filename": "a",
    "id": "0",
  },
  Object {
    "basename": "c",
    "filename": "c",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 7`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(c)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(c)\\",\\"level\\":\\"debug\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 8`] = `
Array [
  Object {
    "level": "silly",
    "message": "removeSound(c)",
    "sounds": Array [
      Object {
        "basename": "a",
        "filename": "a",
        "id": "0",
      },
      Object {
        "basename": "b",
        "filename": "b",
        "id": "1",
      },
    ],
    Symbol(level): "silly",
    Symbol(message): "{\\"message\\":\\"removeSound(c)\\",\\"sounds\\":[{\\"filename\\":\\"a\\",\\"id\\":\\"0\\",\\"basename\\":\\"a\\"},{\\"filename\\":\\"b\\",\\"id\\":\\"1\\",\\"basename\\":\\"b\\"}],\\"level\\":\\"silly\\"}",
  },
]
`;

exports[`modules -> sounds removeSound removes file and re-sorts remaining items 9`] = `
Array [
  Object {
    "basename": "a",
    "filename": "a",
    "id": "0",
  },
  Object {
    "basename": "b",
    "filename": "b",
    "id": "1",
  },
]
`;

exports[`modules -> sounds removeSound unknown file, no action 1`] = `
Array [
  Object {
    "level": "debug",
    "message": "removeSound(f)",
    Symbol(level): "debug",
    Symbol(message): "{\\"message\\":\\"removeSound(f)\\",\\"level\\":\\"debug\\"}",
  },
]
`;
